### ISSUE TITLE
Cast coercion from byte to int reported redundant

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#54447]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Cast-coercion-from-byte-to-int-reported-/1461353)._

---
[severity:Other]

```cs
int RGBdist2(Color i, Color s)
{
return (((int)(i.R) - (int)(s.R)) * ((int)(i.R) - (int)(s.R)) +
            ((int)(i.G) - (int)(s.G)) * ((int)(i.G) - (int)(s.G)) +
            ((int)(i.B) - (int)(s.B)) * ((int)(i.B) - (int)(s.B)));
}
```

IDE0047 but bytes are unsigned.  Without casting before subtraction leads to incorrect results.
Chasing why these are incorrect IDE0047's, or worse following the potential fixes, takes time and leads to doubt.

Microsoft Visual Studio Community 2019
Version 16.10.2
VisualStudio.16.Release/16.10.2+31410.357
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

Visual C++ 2019   00435-60000-00000-AA110
Microsoft Visual C++ 2019

---
#### scott barnes on 6/26/2021, 04:59 AM: 

<p>I wish to withdraw this reported problem.  The IDE0047 is for the outermost set of parentheses.</p>

#### scott barnes on 6/26/2021, 05:08 AM: 

<p>I wish to withdraw my withdrawal.  Removing the outermost parentheses leads back to the reported problem.</p>

#### scott barnes on 6/26/2021, 05:44 AM: 

<p>I wish to withdraw my withdrawal withdrawal and withdraw this reported problem.  The magic coercion happens during but before the subtraction.<br>
byte i = 1;<br>
byte s = 255;<br>
var d = i - s; // var is int<br>
byte b = i - s; // CS0266 can’t convert int to byte</p>

#### Feedback Bot on 6/27/2021, 08:32 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by Youssef1313** [comment]
Related to #50043

**Comment by CyrusNajmabadi** [comment]
I believe the customer is mistaken here.  Specifically: IDE0047 but bytes are unsigned. Without casting before subtraction leads to incorrect results.

This is not true.  C# does not define unsigned subtraction for bytes.  Instead, it always widens these bytes to ints, and does the subtraction there.

These casts are redundant and can be removed safely.
